vernac/vernacinterp.ml

```diff
@@ -215,12 +215,36 @@ let interp_entry ?(verbosely=true) ~st entry =
   Vernacstate.unfreeze_full_state st;
   interp_gen ~verbosely ~st ~interp_fn:interp_control entry
 
-let fs_intern dp =
-  let file = Loadpath.try_locate_absolute_library dp in
-  Feedback.feedback @@ Feedback.FileDependency (Some file, Names.DirPath.to_string dp);
-  let res = Library.intern_from_file file, ("file", file) in
-  Feedback.feedback @@ Feedback.FileLoaded (Names.DirPath.to_string dp, file);
-  res
+module Intern = struct
+  let error_unmapped_dir qid =
+    let prefix, _ = Libnames.repr_qualid qid in
+    CErrors.user_err
+      Pp.(seq [ str "Cannot load "; Libnames.pr_qualid qid; str ":"; spc ()
+              ; str "no physical path bound to"; spc ()
+              ; Names.DirPath.print prefix; fnl ()
+              ])
+
+  let error_lib_not_found dir =
+    let vos = !Flags.load_vos_libraries in
+    let vos_msg = if vos then [Pp.str " (while searching for a .vos file)"] else [] in
+    CErrors.user_err
+      Pp.(seq ([ str "Cannot find library "; Names.DirPath.print dir; str" in loadpath"]@vos_msg))
+
+
+  let fs_intern dp =
+    match Loadpath.locate_absolute_library dp with
+    | Ok file ->
+      Feedback.feedback @@ Feedback.FileDependency (Some file, Names.DirPath.to_string dp);
+      let res = Library.intern_from_file file, ("file", file) in
+      Feedback.feedback @@ Feedback.FileLoaded (Names.DirPath.to_string dp, file);
+      Ok res
+    | Error Loadpath.Error.LibNotFound ->
+      error_lib_not_found dp
+    | Error Loadpath.Error.LibUnmappedDir ->
+      error_unmapped_dir (Libnames.qualid_of_dirpath dp)
+end
+
+let fs_intern = Intern.fs_intern
```

Comment:
vernacinterp probably isn't the right place for this.

I see it's used just below in interp_qed_delayed_proof, but that only runs Qed so either we should have a more specific API that avoids needing the intern argument, or we could pass a dummy intern which asserts false when called.

Reply:
>a more specific API that avoids needing the intern argument

https://github.com/coq/coq/pull/19194

Reply:
Indeed I didn't want to change the API and thus do another round of overlays for this small change, but I see what you mean.

edit: just saw the PR, thanks!

Reply:
Would you mind if we do the move of `fs_intern` to another module in a different PR?

It is not so clear to me where.

Reply:
ok

Reply:
maybe it would fit in loadpath

Reply:
If we want to reuse loadpath among the checker, coqdoc, coqdep , maybe that wouldn't work then.

I have been careful in these years not to try to add these deps to loadpath, so it could indeed be moved to `boot` for example.

As of today it only depends on :
- `Names` + Libnames.qualid
- Mltop for registering ml paths (that could be easily replaced by a callback)